src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1692,6 +1692,11 @@ void ObjectMonitor::wait(jlong millis, bool interruptible, TRAPS) {
       return;
     }
   }
+  if (interruptible) {
+    JavaThreadStatus state = millis != 0 ? JavaThreadStatus::IN_OBJECT_WAIT_TIMED
+                                         : JavaThreadStatus::IN_OBJECT_WAIT;
+    java_lang_Thread::set_thread_status(current->threadObj(), state);
+  }
```

Comment:
It does look a bit hacky. How about moving JavaThreadInObjectWaitState here and change the constructor instead to receive interruptible as argument. Something like:

```
JavaThreadInObjectWaitState(JavaThread *java_thread, bool interruptible, bool timed) :
    JavaThreadStatusChanger(java_thread, !interruptible ? java_lang_Thread::get_thread_status(java_thread->threadObj()) : timed ? JavaThreadStatus::IN_OBJECT_WAIT_TIMED : JavaThreadStatus::IN_OBJECT_WAIT) {
```

Reply:
Good suggestion, thanks. I was also thinking about this. So, will give it a try now.

Reply:
Re-fixed with the suggestion above. The mach5 tiers 1-6 are all passed.